In [1]:
import pandas as pd
df = pd.read_csv('bridges.csv')
df.shape

(108, 13)

In [2]:
df.head(10)

,Id,river,location,erected,purpose,length,lanes,clear-g,t-or-d,material,span,rel-l,type
0,E1,M,3.0,1818,HIGHWAY,NaN,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
1,E2,A,25.0,1819,HIGHWAY,1037.0,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
2,E3,A,39.0,1829,AQUEDUCT,NaN,1.0,N,THROUGH,WOOD,NaN,S,WOOD
3,E5,A,29.0,1837,HIGHWAY,1000.0,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
4,E6,M,23.0,1838,HIGHWAY,NaN,2.0,N,THROUGH,WOOD,NaN,S,WOOD
5,E7,A,27.0,1840,HIGHWAY,990.0,2.0,N,THROUGH,WOOD,MEDIUM,S,WOOD
6,E8,A,28.0,1844,AQUEDUCT,1000.0,1.0,N,THROUGH,IRON,SHORT,S,SUSPEN
7,E9,M,3.0,1846,HIGHWAY,1500.0,2.0,N,THROUGH,IRON,SHORT,S,SUSPEN
8,E10,A,39.0,1848,AQUEDUCT,NaN,1.0,N,DECK,WOOD,NaN,S,WOOD
9,E11,A,29.0,1851,HIGHWAY,1000.0,2.0,N,THROUGH,WOOD,MEDIUM,S,WOOD


In [3]:
df.describe()

,location,erected,length,lanes
count,107.000000,108.000000,81.000000,92.000000
mean,25.978505,1905.314815,1567.469136,2.630435
std,13.665871,37.173664,747.491523,1.164657
min,1.000000,1818.000000,804.000000,1.000000
25%,15.500000,1884.000000,1000.000000,2.000000
50%,27.000000,1903.000000,1300.000000,2.000000
75%,37.500000,1928.000000,2000.000000,4.000000
max,52.000000,1986.000000,4558.000000,6.000000


checking for missing values

In [4]:
df.isna().sum()

,0
Id,0
river,0
location,1
erected,0
purpose,0
length,27
lanes,16
clear-g,2
t-or-d,6
material,2


For the homework, we're only concerned with the length column. Looking at that column, the mean is around 1567.47 while the median is 1300.0. Judging by the distribution of the lengths in the data, I'm more inclinded to used the median to replace the missing values rather than the mean. While 1567.47 isn't close to the maximum value, so far, there is only 2 values in the data over 1300 which leads me to believe theyre skewing the average a little higher. Replacing the missing values will probably give me a very unrealistic outlook, while using the median will skew the data closer to the already established data points.

In [6]:
fill_value = df['length'].median()
df['length'] = df['length'].fillna(fill_value)

1300.0
